sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RequestTimeline.java

```diff
@@ -183,9 +183,9 @@ public Event(final String name, final Instant from, final Instant to) {
             }
 
             if (duration != null) {
-                this.durationInMicroSec = duration.toNanos()/1000L;
+                this.durationInMilliSecs = (double)(duration.toNanos()) / (1000d * 1000d);
```

Comment:
do not think we need the double casting since we are using 1000**d** here

Reply:
As far as compiler inference, you may be correct. But toNanos() returns a long. The cast calls attention to the fact that toNanos() is not long. If the team consensus is to remove the cast, I am glad to oblige. However, as an engineer looking at someone else's code, I would welcome this clarity of intent. Please let me know if I should move forward with the change.

Reply:
make sense, let us keep it then